nixos/tests/systemd-userdbd.nix

```diff
@@ -0,0 +1,32 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "systemd-userdbd";
+  nodes.machine = { config, pkgs, ... }: {
+    services.userdbd.enable = true;
+
+    users.users.test-user-nss = {
+      isNormalUser = true;
+    };
+
+    environment.etc."userdb/test-user-dropin.user".text = builtins.toJSON {
+      userName = "test-user-dropin";
+    };
+
+    environment.systemPackages = with pkgs; [ libvarlink ];
+  };
+  testScript = ''
+    import json
+    from shlex import quote
+
+    def getUserRecord(name):
+      Interface = "unix:/run/systemd/userdb/io.systemd.Multiplexer/io.systemd.UserDatabase"
+      payload = json.dumps({
+        "service": "io.systemd.Multiplexer",
+        "userName": name
+      })
+      return json.loads(machine.succeed(f"varlink call {Interface}.GetUserRecord {quote(payload)}"))
```

Comment:
Can you add some tests using`getent` here, to make sure the NSS side of things also works?

Or could this potentially be folded into `nixos/tests/nscd.nix`, having one machine config with and one without userdbd enabled?

Reply:
Hmmh, we could hide `/etc/userdb/test-user-dropin.user` for the getent process with libredirect, like we hide `/etc/hosts` (so these lookups can only succeed if they go via ns(n)cd).

But it's a bit ugly indeed, and the current `whoami` check is probably fine to ensure it gets the user id somehow?! WDYT?

Reply:
> Hmmh, we could hide `/etc/userdb/test-user-dropin.user` for the getent process with libredirect, like we hide `/etc/hosts` (so these lookups can only succeed if they go via ns(n)cd).

Yea but all lookup using `nss-systemd` has to go through `nscd` which can see all files anyway :p

I think for userdbd this simple check is enough.